### ISSUE TITLE
Nested belongs-to validators gives error

### ISSUE THREAD

**Issue by caalle** [offirgolan/ember-cp-validations#162]
Hey,

I've got nested `belongs-to` validators and the relationship does not have to exist. Eg.

```javascript
//model/foo.js
var Validations = buildValidations({
    bar: validator('belongs-to')
});

export default DS.Model.extend(Validations, {
    name: DS.attr(),
    bar: DS.belongsTo('bar')  
});

//model/bar.js
var Validations = buildValidations({
    baz: validator('belongs-to')
});

export default DS.Model.extend(Validations, {
    name: DS.attr(),
    baz: DS.belongsTo('baz')
});

//model/baz.js
var Validations = buildValidations({
    name: validator('presence', true)
});

export default DS.Model.extend(Validations, {
    name: DS.attr()  
});
```

Creating a `foo` model and calling validate without adding `bar` and `baz` now triggers and error.

```javascript

let model = this.store.createRecord('foo');
model.validate({}, true).then(({model, validations}) => {
    console.log(model, validations); 
});
``` 

gives

```
ember.debug.js:32127 Error: Assertion Failed: Cannot call get with 'validations' on an undefined object.
    at new Error (native)
    at Error.EmberError (http://localhost:4202/assets/vendor.js:26688:21)
    at assert (http://localhost:4202/assets/vendor.js:16447:13)
    at Object.assert (http://localhost:4202/assets/vendor.js:26452:34)
    at get (http://localhost:4202/assets/vendor.js:30911:22)
    at http://localhost:4202/assets/vendor.js:69404:20
    at tryCatch (http://localhost:4202/assets/vendor.js:63616:14)
    at invokeCallback (http://localhost:4202/assets/vendor.js:63631:15)
    at publish (http://localhost:4202/assets/vendor.js:63599:9)
    at http://localhost:4202/assets/vendor.js:42155:7
``` 

I've traced it to the `belongs-to` validator and temporarily added a check for the model variable:

```javascript

export default Base.extend({
  validate(value) {
    if (value) {
      if (canInvoke(value, 'then')) {
        return value.then(model => {
            return model && get(model, 'validations');
        });
      }
      return get(value, 'validations');
    }

    return true;
  }
});

```
Do you have any input on how to solve this? Is there any other way to define the nested `belongs-to`?

Cheers,
Carl

**Comment by offirgolan** [comment]
Thanks for opening up this issue! The error makes sense, currently, its assumed that if you have validations for an object, then that object should exist. Ill write up a test for this and get a fix out when I get the chance. In the mean time, you can do something as such with the current public APIs

```js
//model/bar.js
var Validations = buildValidations({
    baz: validator('belongs-to', {
     disabled() { return Ember.isNone(this.get('model.baz')); }
   })
});

export default DS.Model.extend(Validations, {
    name: DS.attr(),
    baz: DS.belongsTo('baz')
});
```

**Comment by caalle** [comment]
Thanks for the quick reply! Using `disabled` works like a charm.

**Comment by offirgolan** [comment]
Fixed with `v2.7.2` :smile_cat:

**Comment by caalle** [comment]
That was fast. Thanks for a great addon!
